File: IsometricShooter/Scripts/CS/Configuration/NpcConf.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class NpcConf : MonoBehaviour {

	public float speed = 5.0f; //Speed at move movement
	public float attackDistance = 10.0f; //Distance within the NPC attacks the player
	public float sightDistance = 15.0f; //Distance within the NPC sees the player
	public float holdDistance = 2.0f; //Distance within the NPC holds position while attacking the target
	public GameObject onHitEffect; //Effect on hit by projectile
	public Effect onHitEffectType; //Type of effect on hit by projectile
	public GameObject onDieEffect; //Effect on die by projectile
	public Effect onDieEffectType; //Type of effect on die by projectile
	public AudioClip sfxHit; //Audio to be played in case of bullet hit
	public AudioClip sfxDie; //Audio to be played in case of death
	public AudioClip sfxScream; //Audio to be played in case of scream
	public AudioClip sfxCall; //Audio to be played in case of call
	public int getXpOnKill = 10; //XP the player gains on kill of this NPC
	public Color getXpColor = Color.yellow; //Color of spawned XP points
	public GameObject killLoot; //Object spawned as kill loot ("None" = nothing will spawn)
	public Transform target; //target reference (player)
	public AiAction action; //Current action of NPC
	public List<Transform> patrolPointList; //list of patrol points
	public List<UnitSpawn> unitSpawnList; //list of spawn points for units
	
}

File: IsometricShooter/Scripts/JS/Configuration/NpcConf.js
#pragma strict

var speed : float = 5.0; //Speed at move movement
var attackDistance : float = 10.0; //Distance within the NPC attacks the player
var sightDistance : float = 15.0; //Distance within the NPC sees the player
var holdDistance : float = 2.0; //Distance within the NPC holds position while attacking the target
var onHitEffect : GameObject; //Effect on hit by projectile
var onHitEffectType : Effect; //Type of effect on hit by projectile
var onDieEffect : GameObject; //Effect on die by projectile
var onDieEffectType : Effect; //Type of effect on die by projectile
var sfxHit : AudioClip; //Audio to be played in case of bullet hit
var sfxDie : AudioClip; //Audio to be played in case of death
var sfxScream : AudioClip; //Audio to be played in case of scream
var sfxCall : AudioClip; //Audio to be played in case of call
var getXpOnKill : int = 10; //XP the player gains on kill of this NPC
var getXpColor : Color = Color.yellow; //Color of spawned XP points
var killLoot : GameObject; //Object spawned as kill loot ("None" = nothing will spawn)
var target : Transform; //target reference (player)
var action : AiAction; //Current action of NPC
var patrolPointList : List.<Transform>; //list of patrol points
var unitSpawnList : List.<UnitSpawn>; //list of spawn points for units